contracts/test/integration/IntegrationTestVoltV2.t.sol

```diff
@@ -0,0 +1,375 @@
+        assertEq(volt.delegates(address(this)), address(0xFFF));
+    }
+
+    function testMintSuccessMinter(uint256 voltToMint) public {
+        vm.prank(MainnetAddresses.GOVERNOR);
+        core.grantMinter(address(this));
+
+        volt.mint(address(0xFFF), voltToMint);
+
+        assertEq(volt.totalSupply(), voltToMint);
+        assertEq(volt.balanceOf(address(0xFFF)), voltToMint);
+    }
+
+    function testMintFailureUnauthorized() public {
+        vm.expectRevert("CoreRef: Caller is not a minter");
+        volt.mint(address(0xFFF), 1e18);
+    }
+
+    function testMintFailToVoltContract() public {
+        vm.startPrank(MainnetAddresses.GOVERNOR);
+        vm.expectRevert("Volt: cannot transfer to the volt contract");
+        volt.mint(address(volt), 1e18);
+        vm.stopPrank();
+    }
+
+    function testMintFailZeroAddress() public {
+        vm.startPrank(MainnetAddresses.GOVERNOR);
+        vm.expectRevert("Volt: cannot transfer to the zero address");
+        volt.mint(address(0), 1e18);
+        vm.stopPrank();
+    }
+
+    function testBurn(uint256 voltToBurn) public {
+        vm.prank(MainnetAddresses.GOVERNOR);
+        volt.mint(address(this), voltToBurn);
+        volt.burn(voltToBurn);
+
+        assertEq(volt.totalSupply(), 0);
+        assertEq(volt.balanceOf(address(this)), 0);
+    }
+
+    function testBurnFail() public {
+        vm.prank(MainnetAddresses.GOVERNOR);
+        volt.mint(address(this), 1e18);
+
+        vm.expectRevert("Volt: burn amount exceeds balance");
+        volt.burn(2e18);
+    }
+
+    function testBurnFailZeroAddress() public {
+        vm.startPrank(address(0));
+        vm.expectRevert("Volt: cannot burn from the zero address");
+        volt.burn(1e18);
+        vm.stopPrank();
+    }
+
+    function testBurnFrom(uint256 voltToBurn) public {
+        vm.assume(voltToBurn < type(uint256).max); // to make sure we don't run into infinite approval counter example
+        address from = address(0xFFF);
+        vm.prank(MainnetAddresses.GOVERNOR);
+        volt.mint(from, voltToBurn);
+
+        vm.prank(from);
+        volt.approve(address(this), voltToBurn);
+        assertEq(volt.allowance(from, address(this)), voltToBurn);
+
+        volt.burnFrom(from, voltToBurn);
+
+        assertEq(volt.balanceOf(from), 0);
+        assertEq(volt.allowance(from, address(this)), 0);
+        assertEq(volt.totalSupply(), 0);
+    }
+
+    function testBurnFromFailAddressZero() public {
+        address from = address(0);
+
+        vm.prank(from);
+        volt.approve(address(this), 1e18);
+        assertEq(volt.allowance(from, address(this)), 1e18);
+
+        vm.expectRevert("Volt: cannot burn from the zero address");
+        volt.burnFrom(from, 1e18);
+    }
+
+    function testBurnFromInfiniteApproval(uint256 voltToBurn) public {
+        address from = address(0xFFF);
+        vm.prank(MainnetAddresses.GOVERNOR);
+        volt.mint(from, voltToBurn);
```

Comment:
on a mint, we need to verify the user has been properly check pointed in the voting system. that should be verified along with the total supply incrementing.